% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\title{GGplot}
\author{Ehsan Khedive}
\date{04/25/2022}

\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={GGplot},
  pdfauthor={Ehsan Khedive},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\begin{document}
\maketitle

\hypertarget{importing-and-handling-summary-of-rna-seq-data}{%
\subsection{1. Importing and handling summary of RNA-Seq
data}\label{importing-and-handling-summary-of-rna-seq-data}}

Here I imported a summary of RNA-seq data using \texttt{read.delim} and
then converted Region to factor using \texttt{as.factor}. Finally I
printed the structure and first 5 rows of the data and selected the
chromosome 21 for graphing purposes.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ggplot2)}
\NormalTok{RNAseq }\OtherTok{=} \FunctionTok{read.delim}\NormalTok{ (}\StringTok{"SlidingWindow.txt"}\NormalTok{)}
\FunctionTok{str}\NormalTok{(RNAseq)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 'data.frame':    28581 obs. of  5 variables:
##  $ Chrom     : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ Pos       : int  251 501 751 1001 1251 1501 1751 2001 2251 2501 ...
##  $ Region    : chr  "intergenic" "intergenic" "intergenic" "intergenic" ...
##  $ RCM       : num  1 1 1 1 1 1 1 1 1 1 ...
##  $ Enrichment: num  0 0 0 0 0 0 0 0 0 0 ...
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{RNAseq}\SpecialCharTok{$}\NormalTok{Region }\OtherTok{=} \FunctionTok{as.factor}\NormalTok{(RNAseq}\SpecialCharTok{$}\NormalTok{Region)}
\NormalTok{RNAseq}\SpecialCharTok{$}\NormalTok{Region }\OtherTok{=} \FunctionTok{factor}\NormalTok{(RNAseq}\SpecialCharTok{$}\NormalTok{Region, }\AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"genic"}\NormalTok{, }\StringTok{"intergenic"}\NormalTok{))}
\FunctionTok{str}\NormalTok{(RNAseq)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 'data.frame':    28581 obs. of  5 variables:
##  $ Chrom     : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ Pos       : int  251 501 751 1001 1251 1501 1751 2001 2251 2501 ...
##  $ Region    : Factor w/ 2 levels "genic","intergenic": 2 2 2 2 2 2 2 2 2 2 ...
##  $ RCM       : num  1 1 1 1 1 1 1 1 1 1 ...
##  $ Enrichment: num  0 0 0 0 0 0 0 0 0 0 ...
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{RNAseq[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Chrom  Pos     Region RCM Enrichment
## 1     1  251 intergenic   1          0
## 2     1  501 intergenic   1          0
## 3     1  751 intergenic   1          0
## 4     1 1001 intergenic   1          0
## 5     1 1251 intergenic   1          0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{RNAseq\_chrom21 }\OtherTok{=}\NormalTok{ RNAseq[}\FunctionTok{which}\NormalTok{(RNAseq}\SpecialCharTok{$}\NormalTok{Chrom}\SpecialCharTok{==}\DecValTok{21}\NormalTok{),]}
\end{Highlighting}
\end{Shaded}

\hypertarget{creating-the-plot-and-layout}{%
\subsection{2. Creating the plot and
layout}\label{creating-the-plot-and-layout}}

In this step I laid out the plot step by step and then graphed the data
using \texttt{ggplot} command and named it \textbf{singlePlot}.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{singlePlot }\OtherTok{=} \FunctionTok{ggplot}\NormalTok{ (}\AttributeTok{data=}\NormalTok{RNAseq\_chrom21, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Pos, }\AttributeTok{y=}\NormalTok{RCM))}
\FunctionTok{plot}\NormalTok{ (RNAseq\_chrom21}\SpecialCharTok{$}\NormalTok{Pos, RNAseq\_chrom21}\SpecialCharTok{$}\NormalTok{RCM)}
\end{Highlighting}
\end{Shaded}

\includegraphics{GGplot_files/figure-latex/unnamed-chunk-2-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{singlePlot }\OtherTok{=} \FunctionTok{ggplot}\NormalTok{ (}\AttributeTok{data=}\NormalTok{RNAseq\_chrom21, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Pos, }\AttributeTok{y=}\NormalTok{RCM)) }\SpecialCharTok{+} \FunctionTok{geom\_point}\NormalTok{()}
\NormalTok{singlePlot }\OtherTok{=} \FunctionTok{ggplot}\NormalTok{ (}\AttributeTok{data=}\NormalTok{RNAseq\_chrom21, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Pos, }\AttributeTok{y=}\NormalTok{RCM, }\AttributeTok{color=}\NormalTok{Region)) }\SpecialCharTok{+} \FunctionTok{geom\_point}\NormalTok{()}
\NormalTok{singlePlot }\SpecialCharTok{+} \FunctionTok{scale\_color\_manual}\NormalTok{(}\AttributeTok{values=}\FunctionTok{c}\NormalTok{(}\StringTok{"red"}\NormalTok{, }\StringTok{"black"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{GGplot_files/figure-latex/unnamed-chunk-2-2.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{singlePlot2 }\OtherTok{=} \FunctionTok{ggplot}\NormalTok{ (}\AttributeTok{data=}\NormalTok{RNAseq\_chrom21, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Pos, }\AttributeTok{y=}\NormalTok{RCM, }\AttributeTok{color=}\NormalTok{Enrichment)) }\SpecialCharTok{+} \FunctionTok{geom\_point}\NormalTok{()}
\NormalTok{singlePlot2}
\end{Highlighting}
\end{Shaded}

\includegraphics{GGplot_files/figure-latex/unnamed-chunk-2-3.pdf}

\hypertarget{customizing-plots}{%
\subsection{3. Customizing plots}\label{customizing-plots}}

Customization was done using \texttt{viridis} library to create a new
color theme on the chart. Chart axis also was added using \texttt{xlab}
and \texttt{ylab} commands.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(viridis)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: package 'viridis' was built under R version 4.1.3
\end{verbatim}

\begin{verbatim}
## Loading required package: viridisLite
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{singlePlot2 }\SpecialCharTok{+} \FunctionTok{scale\_color\_viridis}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\includegraphics{GGplot_files/figure-latex/unnamed-chunk-3-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggplot}\NormalTok{ (}\AttributeTok{data=}\NormalTok{RNAseq\_chrom21, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Pos, }\AttributeTok{y=}\NormalTok{RCM, }\AttributeTok{color=}\NormalTok{Enrichment)) }\SpecialCharTok{+} \FunctionTok{geom\_point}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{scale\_color\_viridis}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{scale\_y\_log10}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\includegraphics{GGplot_files/figure-latex/unnamed-chunk-3-2.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggplot}\NormalTok{ (}\AttributeTok{data=}\NormalTok{RNAseq\_chrom21, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Pos}\SpecialCharTok{/}\DecValTok{1000}\NormalTok{, }\AttributeTok{y=}\NormalTok{RCM, }\AttributeTok{color=}\NormalTok{Enrichment)) }\SpecialCharTok{+} \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{size=}\FloatTok{0.5}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{scale\_color\_viridis}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{scale\_y\_log10}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{xlab}\NormalTok{(}\StringTok{"Nucleotide Position (kb)"}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{ylab}\NormalTok{(}\StringTok{"Read Count Per Million"}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{theme\_bw}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\includegraphics{GGplot_files/figure-latex/unnamed-chunk-3-3.pdf}

\hypertarget{creating-panel-chart-for-all-chromosomes}{%
\subsection{4. Creating panel chart for all
chromosomes}\label{creating-panel-chart-for-all-chromosomes}}

Using \texttt{facet\_wrap} option and defining chromosome number in this
option I creted a panel chart of 7 rows for all of the chromosomes.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggplot}\NormalTok{ (}\AttributeTok{data=}\NormalTok{RNAseq, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Pos}\SpecialCharTok{/}\DecValTok{1000}\NormalTok{, }\AttributeTok{y=}\NormalTok{RCM, }\AttributeTok{color=}\NormalTok{Enrichment)) }\SpecialCharTok{+} \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{size=}\FloatTok{0.5}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{scale\_color\_viridis}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{scale\_y\_log10}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{xlab}\NormalTok{(}\StringTok{"Nucleotide Position (kb)"}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{ylab}\NormalTok{(}\StringTok{"Read Count Per Million"}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{theme\_bw}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{scale\_x\_continuous}\NormalTok{ (}\AttributeTok{breaks=}\FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{75}\NormalTok{,}\DecValTok{150}\NormalTok{)) }\SpecialCharTok{+} \FunctionTok{facet\_wrap}\NormalTok{(}\SpecialCharTok{\textasciitilde{}}\NormalTok{Chrom, }\AttributeTok{nrow=}\DecValTok{7}\NormalTok{) }
\end{Highlighting}
\end{Shaded}

\includegraphics{GGplot_files/figure-latex/unnamed-chunk-4-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#pdf ("multiplot.pdf")}
\CommentTok{\#multiplot  }
\CommentTok{\#dev.off()}
\end{Highlighting}
\end{Shaded}


\end{document}
